elastic-stack/elasticsearch/plugins.sls

```diff
@@ -5,15 +5,19 @@ include:
   - .install
 
 {% for plugin in salt.pillar.get('elastic_stack:elasticsearch:plugins', []) %}
-install_{{ plugin.name }}_plugin:
+remove_{{ plugin.name }}_plugin:
   cmd.run:
     - name: /usr/share/elasticsearch/bin/elasticsearch-plugin remove {{ plugin.get('location', plugin.name) }}
-    - onchanges: install_elasticsearch
-    - watch_in:
-      - service: elasticsearch_service
+    - onchanges:
+      - pkg: install_elasticsearch
+    - require_in:
+      - cmd: install_{{ plugin.name }}_plugin
```

Comment:
I think this looks good, maybe just add an `onlyif` statement so it doesn't run when the plugin hasn't yet been installed.

Reply:
I've added an `onlyif` in 488d5c56d6106a5e611d4738cd3952e4ed2509c0

Reply:
... But wondering how you feel about the `test -e` on the `.jar` file, since the state used to use the following test, in the install state:

`[ $(/usr/share/elasticsearch/bin/elasticsearch-plugin list | grep {{ plugin.name }} | wc -l) -eq 1 ]`

Reply:
I'm not dogmatic about it, that was just the approach I had taken before. Happy to do whatever works.